src/sentry/auth/helper.py

```diff
@@ -56,62 +57,14 @@
 ERR_INVALID_IDENTITY = _("The provider did not return a valid user identity.")
 
 
-class RedisBackedState:
-    # Expire the pipeline after 10 minutes of inactivity.
-    EXPIRATION_TTL = 10 * 60
-
-    def __init__(self, request):
-        self.__dict__["request"] = request
-
-    @property
-    def _client(self):
-        return clusters.get("default").get_local_client_for_key(self.auth_key)
+class RedisBackedState(PipelineSessionStore):
```

Comment:
Hmm, I thought there was a way to use this without having to extend it. Maybe not?

Reply:
The complications are:

1. The original code constructed `RedisBackedState` in two places: `get_for_request_` and `__init__`. Even if we're overriding it in `get_for_request_`, `Pipeline.__init__` would still be assigning a `RedisSessionStore`. So they need to be unified somehow.
2. The original code was using different prefixes from `RedisSessionStore`. It *might* be safe to just change that, but I'd assume it risks breaking the flow is a user is midway through at the time the change is deployed. So the extra attributes let us preserve exact behavior.
3. If we go forward with `RedisSessionStore` no longer allowing dynamic attributes, this is the best place for `flow = redis_property("flow")` to live.

I would support renaming it; only left it alone to minimize code movement.